citations-tool/tool/src/webapp/vm/citation/_listNestedCitations.vm

```diff
@@ -39,6 +39,24 @@
 		#set( $addH1SubsectionButtonId = 'addH1SubsectionButton' + $nestedSection.getLocation() )
 		#set( $buttonsDiv = 'buttonsDiv' + $nestedSection.getLocation() )
 		#set( $num = $num + 1)
+		#set( $citationNo = 0)
```

Comment:
mmm I think it belongs more in the vm because it's not *a* simple value you get in the Java and pop in the context - it's calculating the number of citations for each h1 that it's iterating through and and showing that number - it makes more sense to me for this code to go during the iteration.
you're right about it not being as debuggable as in Java but you could say that about anything in the vm